File: lib/models/ingredient.dart
class Ingredient {
  const Ingredient({required this.id, required this.name, required this.calories, required this.referenceAmount, required this.unit});
  final String id;
  final String name;
  final double calories;
  final double referenceAmount;
  final String unit;
  double caloriesFor(double amount) => referenceAmount <= 0 ? 0 : calories * amount / referenceAmount;
}

File: app/src/main/java/com/sam/caloriestreak/domain/model/Ingredient.kt
package com.sam.caloriestreak.domain.model

enum class IngredientUnit { GRAM, MILLILITRE, PIECE, SLICE, TABLESPOON, TEASPOON, CUP, CUSTOM }

data class Ingredient(
    val id: String,
    val name: String,
    val brand: String? = null,
    val calories: Double,
    val referenceAmount: Double,
    val referenceUnit: IngredientUnit,
    val favorite: Boolean = false,
    val archived: Boolean = false
) {
    fun caloriesFor(amount: Double): Double =
        if (referenceAmount <= 0.0) 0.0 else calories * amount / referenceAmount
}
